src/note.rs

```diff
@@ -226,6 +227,25 @@ impl Note {
         }
     }
 
+    /// Reconstruct a Note from its individual fields.
+    pub fn from_fields(
+        note_type: NoteType,
+        value_commitment: JubJubExtended,
+        nonce: JubJubScalar,
+        stealth_address: StealthAddress,
+        pos: u64,
+        encrypted_data: [u8; ENCRYPTED_DATA_SIZE],
+    ) -> Self {
+        Note {
+            note_type,
+            value_commitment,
+            nonce,
+            stealth_address,
+            pos,
+            encrypted_data,
+        }
+    }
```

Comment:
I would close this PR, a method such this one makes no sense. If you need to create a struct passing all the fields means you basically creates the struct directly, so:
```rust
let my_note = Note {
            note_type,
            value_commitment,
            nonce,
            stealth_address,
            pos,
            encrypted_data,
        };
```

Reply:
To be clear, this won't be the approach I would take, since it would means everything would be in write / read mode for the consumer. In term of design API we need to understand why we need specifically this, and figure out an alternative.

Reply:
This approach also would not work, since none of the fields are public. Hence why I have the method.

I think the issue/commit explains why we need it (decoding from protocol buffers), without this I'd need to write a lot more code in order to actually decode a note.

Reply:
After a brief discussion on call, the conclusion was to not expose any of these fields directly and instead using the `Read/Write` traits to write all the data onto a newly decoded `Note`.